### DISCUSSION TITLE
how can vscode-go support group import

### DISCUSSION THREAD

**Discussion by wondar-chan** [golang/vscode-go#2004] [Help] [answered]
i am using goland and vscode for develop. file watcher of goland support group import for 3 groups: go, third party, own. But when it comes to vscode, i set the format flag as: "-local $local(my own package name)", and use goimports as format tool but it won't work for me, when i format file, the import was sorted by default rule. i am curious. Besides, both able and disable of the language server have no difference

**Comment by butuzov** [comment, marked as answer]
What you can do is try to debug why it's happening. Is `goimports` available from terminal? Can you run `goimports` command  and see file changed? 

btw, did you consider alternatives? I am using `gofumpt` instead of `goimports` and you can turn its support by providing an appropriate bool flag to `gopls.gofumpt` setting.

```json
"go.useLanguageServer": true,
"go.formatTool": "gofumpt",
"go.formatFlags": [],
"gopls": {
	"formatting.gofumpt": true,
},
```

**Reply by hyangah** [reply]
Thanks for the answer @butuzov 

@wondar-chan FYI, `go.formatFlags` is not applicable when using the language server. When using the language server, utilize the "gopls" settings. 

`gofumpt` style - https://github.com/golang/vscode-go/blob/master/docs/settings.md#formattinggofumpt 
`goimports -local` style - https://github.com/golang/vscode-go/blob/master/docs/settings.md#formattinglocal

**Reply by wondar-chan** [reply]
Thank you guys @hyangah @butuzov ! It's ok when I set "formatting.local".   https://github.com/golang/vscode-go/blob/master/docs/settings.md#formattinglocal
